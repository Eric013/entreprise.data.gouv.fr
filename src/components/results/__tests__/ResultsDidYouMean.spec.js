import ResultsDidYouMean from '@/components/results/ResultsDidYouMean.vue'
import Vuex from 'vuex'
import { __createMocks as createStoreMocks } from '@/store/index.js'
import { createLocalVue, shallow } from '@vue/test-utils'


const localVue = createLocalVue()
localVue.use(Vuex)
jest.mock('@/store/index.js')
jest.mock('@/components/results/ResultsDidYouMean.vue')

describe('ResultsDidYouMean.vue', () => {
  let storeMocks
  let wrapperDYM

  beforeEach(() => {
    storeMocks = createStoreMocks()

    wrapperDYM = shallow(ResultsDidYouMean, {
      localVue,
      store: storeMocks.store
    })
  })

  test('Computed value didYouMean return the correct getter', () => {
    expect(wrapperDYM.vm.didYouMean).toBe('mock-spellcheck')
  })

  test('Method didYouMeanSearch sets fullText at param value', () => {
    const didYouMean = 'mock-didYouMean'
    wrapperDYM.vm.didYouMeanSearch(didYouMean)
    expect(storeMocks.mutations.setFullText).toHaveBeenCalledWith(storeMocks.state, didYouMean)
  })
  test('Method didYouMeanSearch launch requestSearch', () => {
    const didYouMean = 'mock-didYouMean'
    wrapperDYM.vm.didYouMeanSearch(didYouMean)
    expect(storeMocks.actions.requestSearch).toHaveBeenCalled()
  })
}) 