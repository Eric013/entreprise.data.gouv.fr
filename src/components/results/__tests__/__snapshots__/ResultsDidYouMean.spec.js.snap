// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsDidYouMean.vue Snapshot test Snapshot test 1`] = `<!---->`;
